src/auth/ejabberd_auth_rdbms.erl

```diff
@@ -287,8 +287,13 @@ get_password(LUser, LServer) ->
             Password; %%Plain password
         {selected, [{_Password, PassDetails}]} ->
             case mongoose_scram:deserialize(PassDetails) of
-                {ok, Scram} ->
-                    mongoose_scram:scram_to_tuple(Scram);
+                {ok, Scram} when is_record(Scram, scram) ->
```

Comment:
How about we return the new map from `mongoose_scram:deserialize`?

Reply:
This was possible only after doing rebase with the base branch as deserialize was done there after this PR was created. Now it's fixed.